src/include/catalog/settings_catalog.h

```diff
@@ -0,0 +1,73 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// settings_catalog.h
+//
+// Identification: src/include/catalog/settings_catalog.h
+//
+// Copyright (c) 2015-2017, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+
+#pragma once
+
+#include "catalog/abstract_catalog.h"
+
+namespace peloton {
+namespace catalog {
+
+class SettingsCatalog : public AbstractCatalog {
+public:
+  ~SettingsCatalog();
+
+  // Global Singleton
+  static SettingsCatalog *GetInstance(
+          concurrency::Transaction *txn = nullptr);
```

Comment:
After the first call to GetInstance funciton, we needn't pass a transaction pointer here.